src/web/core/src/main/java/org/geoserver/web/wicket/GeoServerTablePanel.java

```diff
@@ -580,12 +580,10 @@ public PagerDelegate() {
          * Updates the label given the current page and filtering status
          */
         void updateMatched() {
+            size = dataProvider.size();
             fullSize = dataProvider.fullSize();
-            first = first(fullSize);
-            last = last(fullSize);
-            if (dataProvider.getKeywords() != null) {
-                size = dataProvider.size();
-            }
```

Comment:
`dataProvider.getKeywords() != null` checks if there is anything in the filter. Removed the check so that the count gets reset properly when the filter is cleared. Use size (size with filter) instead of fullSize (size without) for calculations (both these are global)